src/libAtomVM/opcodesswitch.h

```diff
@@ -1423,9 +1427,38 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
             case OP_ALLOCATE_HEAP: {
                 int next_off = 1;
                 int stack_need;
-                DECODE_INTEGER(stack_need, code, i, next_off, next_off);
+                if (UNLIKELY(IS_EXTENDED_ALLOCATOR(code, i, next_off))) {
```

Comment:
Could we factor this block with a new macro?

Reply:
Which test covers this case? I'm not sure `stack_need` can be an extended allocator list with OTP24.

Reply:
yes, - c2fe3a35e180b46cba908b3ec1dd930fbbed2cb0